CMakeLists.txt

```diff
@@ -348,13 +348,35 @@ target_link_libraries(core_interface INTERFACE
   Threads::Threads
 )
 
+# Define "sanitize_interface" with flags intended to apply to all libraries and
+# executables to support sanitizers and fuzzing, and a separate
+# "fuzzer_interface" with flags specifically intended to apply to the fuzz test
+# binary. Reason these are separate interfaces is that if -DSANITIZERS=fuzzer is
+# specified, the fuzz test binary should be built with -fsanitize=fuzzer (so it
+# can use libFuzzer's main function), but libraries should be built with
+# -fsanitize=fuzzer-no-link (so they can be linked into other executables that
+# have their own main functions).
 add_library(sanitize_interface INTERFACE)
+add_library(fuzzer_interface INTERFACE)
 target_link_libraries(core_interface INTERFACE sanitize_interface)
 if(SANITIZERS)
+  # Transform list of sanitizers into -fsanitize flags, replacing "fuzzer" with
+  # "fuzzer-no-link" in sanitize_interface flags, and moving "fuzzer" to
+  # fuzzer_interface flags.
+  string(REGEX REPLACE "(^|,)fuzzer($|,)" "\\1fuzzer-no-link\\2" SANITIZE_FLAG "${SANITIZERS}")
+  set(FUZZ_FLAG "")
+  if (NOT "${SANITIZE_FLAG}" STREQUAL "${SANITIZERS}")
+    set(FUZZ_FLAG "-fsanitize=fuzzer")
+    if(NOT BUILD_FOR_FUZZING)
+      message(FATAL_ERROR "Error: Enabling -DSANITIZERS=fuzzer without -DBUILD_FOR_FUZZING=ON is not supported.")
+    endif()
+  endif()
+  set(SANITIZE_FLAG "-fsanitize=${SANITIZE_FLAG}")
```

Comment:
Maybe avoid reusing the `SANITIZE_FLAG` variable?

style-nit: The surrounding code mostly uses `ALL_CAPS` naming for user-faced variables, not for internal ones.